swap_meet/vendor.py

```diff
@@ -1,2 +1,99 @@
+from swap_meet.item import Item 
+"""
+Module that contains class Vendor with attribute inventory and instance methods: add, remove.
+Other instance methods:
+To get items in inventory by Item object attribute: get_by_id, get_by_category, get_best_by_category, get_newest_by_category 
+To swap items by attribute: swap_best_by_category, , swap_newest_by_category. 
+"""
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+        
+    def add(self, item):
+        self.inventory.append(item)
+        if item in self.inventory:
+            return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        if item not in self.inventory:
+            return item
+    
+    def get_by_id(self, search_id):
+        for item in self.inventory:
+            if item.id == search_id:
+                return item
+        return None   
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if not (my_item in self.inventory and their_item in other_vendor.inventory):
+            return False   
```

Comment:
Excellent work :D This is one of the parts with the most work around instance methods and variables.